packages/core/src/StateMachine.ts

```diff
@@ -442,9 +452,53 @@ export class StateMachine<
       State<TContext, TEvent, TResolvedTypesMeta>
     >
   ): State<TContext, TEvent, TResolvedTypesMeta> {
-    const restoredState = isStateConfig(state)
-      ? this.resolveState(state as any)
-      : this.resolveState(State.from(state as any, this.context, this));
+    let restoredState: State<TContext, TEvent, TResolvedTypesMeta>;
+
+    if ('persisted' in state) {
+      const restoredChildren: Record<string, AnyActorRef> = {};
+
+      Object.keys(state.children).forEach((key) => {
```

Comment:
let's rename `key` here to `id` since that's how it's used later on - at least in some places